onebusaway-api-webapp/src/main/java/org/onebusaway/api/actions/api/where/ArrivalsAndDeparturesForLocationAction.java

```diff
@@ -164,7 +168,17 @@ public DefaultHttpHeaders index() throws IOException, ServiceException {
         if (adResult == null) {
             return emptyResponse();
         }
-        return setOkResponse(factory.getResponse(adResult));
+
+        EntryWithReferencesBean<StopsWithArrivalsAndDeparturesV2Bean> response = factory.getResponse(adResult);
+
+        response.getEntry().getArrivalsAndDepartures()
```

Comment:
I get that this is super convenient to do right here, but it feels wrong.  If you moved it into the BeanFactory, couldn't it be re-used?  I could be wrong -- just asking.  

Basically because you decided to do this here, you need to copy this code fragment to N other places, violating the DRY prinicipal